consensus/src/round_manager.rs

```diff
@@ -1107,18 +1109,35 @@ impl RoundManager {
                 return Ok(());
             }
 
-            if order_vote_msg.order_vote().ledger_info().round()
-                > self.block_store.sync_info().highest_ordered_round()
+            let highest_ordered_round = self.block_store.sync_info().highest_ordered_round();
+            let highest_ordered_epoch = self.block_store.sync_info().highest_ordered_cert().epoch();
+            let order_vote_round = order_vote_msg.order_vote().ledger_info().round();
+            let li_digest = order_vote_msg.order_vote().ledger_info().hash();
+            if order_vote_round > highest_ordered_round
+                && order_vote_round < highest_ordered_round + 100
+                && order_vote_msg.epoch() >= highest_ordered_epoch
             {
+                if !self.pending_order_votes.exists(&li_digest) {
+                    let start = Instant::now();
+                    order_vote_msg
+                        .quorum_cert()
+                        .verify(&self.epoch_state().verifier)
+                        .context("[OrderVoteMsg QuorumCert verification failed")?;
+                    counters::VERIFY_MSG
+                        .with_label_values(&["order_vote_qc"])
+                        .observe(start.elapsed().as_secs_f64());
+                    self.order_vote_qc_cache
```

Comment:
why do we need this cache if we already cache the li_digest in pending_order_votes?

Reply:
This is actually caching the verified QC for the li_digest. When the ordered cert is inserted into the block store, and if there is necessity to add QC into the block store, instead of verifying the QC again, I'm just reusing this verified QC. It's a very minor optimization.

Reply:
i got it, but it feels much better to put it inside the pending_order_votes

Reply:
Done.